openvalidators/event.py

```diff
@@ -32,14 +51,14 @@ class EventSchema:
     def from_dict(event_dict: dict, should_log_rewards: bool) -> 'EventSchema':
         """Converts a dictionary to an EventSchema object."""
         rewards = {
-            'dahoas_reward_model': event_dict.get('dahoas_reward_model'),
-            'blacklist_filter': event_dict.get('blacklist_filter'),
-            'nsfw_filter': event_dict.get('nsfw_filter'),
-            'reciprocate_reward_model': event_dict.get('reciprocate_reward_model'),
-            'diversity_reward_model': event_dict.get('diversity_reward_model'),
-            'rlhf_reward_model': event_dict.get('rlhf_reward_model'),
-            'prompt_reward_model': event_dict.get('prompt_reward_model'),
-            'relevance_scoring': event_dict.get('relevance_scoring'),
+            'dahoas_reward_model': event_dict.get(RewardModelType.dahoas.value),
```

Comment:
IMO keys should be coming from the `RewardModelType` class too. This will ensure that naming conventions are kept consistent.

```python
In [28]: event_dict = {'rlhf_reward_model':[1,2,3], 'dahoas_reward_model': [5,3,1], 'nsfw_filter': [69,69,69]}

In [29]: {reward_model.name: event_dict.get(reward_model.value) for reward_model in RewardModelType}
Out[29]: 
{'rlhf': [1, 2, 3],
 'reciprocate': None,
 'dahoas': [5, 3, 1],
 'diversity': None,
 'prompt': None,
 'blacklist': None,
 'nsfw': [69, 69, 69],
 'relevance': None}
```

Reply:
BTW I'm unable to comment on the lines beyond those which were edited in this PR but I'm seeing that the reward/filter models in the event class have `Optional` type hints but no default values.

Reply:
Yeah, I usually go with the approach `Less code = Better code` but for tests I prefer to go with `Explicit code > Implicit code` for clarity on what's going on. On line 120 you can see that I use this very same "Less code" approach , as the rationale of the test is specifically on that. For this particular one I thought 8 LOCs for more clarity woudn't be a big problem

Reply:
Current solution is ok. Future me is kicking off about it because we are susceptible to naming convention drift, which means that dashboard and any analytics layer is going to be brittle w.r.t. schema changes. 

It's not a hill I'm prepared to die on. After all it's only monday 😆